guillotina_amqp/tests/test_api.py

```diff
@@ -114,3 +121,41 @@ async def test_info_task_filtered_response(
         )
         assert status == 200
         assert "job_data" not in resp
+
+
+async def join_amqp_worker():
+    sm = get_state_manager()
+    async for task_id in sm.list():
+        task = TaskState(task_id)
+        try:
+            await task.join()
+        except TaskNotFoundException:
+            pass
+
+
+async def test_example_of_service_spawning_a_task(container_requester, amqp_worker):
+    async with container_requester as requester:
+        # Create item in container with a title
+        resp, status = await requester(
+            "POST",
+            "/db/guillotina/",
+            data=json.dumps({"@type": "Item", "title": "Foo"}),
+        )
+        assert status == 201
+        item_url = "/db/guillotina/" + resp["@name"]
+
+        # Schedule task to clear title of the it
+        resp, status = await requester("POST", item_url + "/@clearTitle")
+        assert status == 200
+        task_id = resp["task_id"]
+
+        # Wait until task has finished
+        await join_amqp_worker()
```

Comment:
Awesome, could we wait only for the specific `task_id` instead of waiting for all of them, indeed something like this

```python
async def join_task(task_id):
    task = TaskState(task_id)
    try:
        await task.join()
    except TaskNotFoundException:
        # task was already finished
        pass
```

should work, am I wrong?